src/main/java/seedu/duke/logic/commands/RouteCommand.java

```diff
@@ -22,13 +22,23 @@ public RouteCommand(String rawMessage) {
     @Override
     public void executeCommand() throws CustomException {
         String [] locations = parser.getLocations();
+        checkLocations(locations);
+        ArrayList<Bus> busOptions = BusData.possibleBuses(locations[0].trim(), locations[1].trim());
+        Ui.printRouteMessage(busOptions);
+    }
+
+    private void checkLocations(String[] locations) throws CustomException {
         if (locations[0].trim().length() == 0 || locations[1].trim().length() == 0) {
             throw new CustomException(ExceptionType.NO_LOCATIONS);
         } else if (locations[0].trim().equalsIgnoreCase(locations[1].trim())) {
+            assert locations[0].trim().length() != 0 : "Location 0 is empty.";
+            assert locations[1].trim().length() != 0 : "Location 1 is empty.";
```

Comment:
```suggestion
            assert !locations[0].isblank() : "Location 0 is empty.";
            assert !locations[1].isblank(): "Location 1 is empty.";
```

Reply:
Looks good. Can consider using .isblank()

Reply:
Oh yes! That's a better way to do it.